src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForXPathDocumentTests.cs

```diff
@@ -0,0 +1,366 @@
+        {
+            VerifyCSharp(@"
+using System.Xml;
+using System.Xml.XPath;
+
+namespace FxCopUnsafeXml
+{
+    public class UseXmlReaderForXPathDocument
+    {
+        public void TestMethod(string path)
+        {
+            XPathDocument doc = new XPathDocument(path);
+        }
+    }
+}
+",
+                GetCA3075XPathDocumentCSharpResultAt(11, 33, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+Imports System.Xml.XPath
+
+Namespace FxCopUnsafeXml
+    Public Class UseXmlReaderForXPathDocument
+        Public Sub TestMethod(path As String)
+            Dim doc As New XPathDocument(path)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XPathDocumentBasicResultAt(8, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInGetShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml.XPath;
+
+class TestClass
+{
+    public XPathDocument Test
+    {
+        get
+        {
+            var xml = """";
+            XPathDocument doc = new XPathDocument(xml);
+            return doc;
+        }
+    }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(11, 33, "get_Test")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Public ReadOnly Property Test() As XPathDocument
+        Get
+            Dim xml = """"
+            Dim doc As New XPathDocument(xml)
+            Return doc
+        End Get
+    End Property
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(8, 24, "get_Test")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInSetShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml.XPath;
+
+class TestClass
+{
+XPathDocument privateDoc;
+public XPathDocument GetDoc
+        {
+            set
+            {
+                if (value == null)
+                {
+                    var xml = """";
+                    XPathDocument doc = new XPathDocument(xml);
+                    privateDoc = doc;
+                }
+                else
+                    privateDoc = value;
+            }
+        }
+}",
+                GetCA3075XPathDocumentCSharpResultAt(14, 41, "set_GetDoc")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml.XPath
+
+Class TestClass
+    Private privateDoc As XPathDocument
+    Public WriteOnly Property GetDoc() As XPathDocument
+        Set
+            If value Is Nothing Then
+                Dim xml = """"
+                Dim doc As New XPathDocument(xml)
+                privateDoc = doc
+            Else
+                privateDoc = value
+            End If
+        End Set
+    End Property
+End Class",
+                GetCA3075XPathDocumentBasicResultAt(10, 28, "set_GetDoc")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInTryBlockShouldGenerateDiagnostic()
```

Comment:
What is special about it being in a `try` block? (I asked the same question below, sorry for the repetition).